common/OpTestSensors.py

```diff
@@ -0,0 +1,182 @@
+#!/usr/bin/python
+# IBM_PROLOG_BEGIN_TAG
+# This is an automatically generated prolog.
+#
+# $Source: op-auto-test/common/OpTestSensors.py $
+#
+# OpenPOWER Automated Test Project
+#
+# Contributors Listed Below - COPYRIGHT 2015
+# [+] International Business Machines Corp.
+#
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+# implied. See the License for the specific language governing
+# permissions and limitations under the License.
+#
+# IBM_PROLOG_END_TAG
+
+#  @package OpTestSensors
+#  Sensors package for OpenPower testing.
+#
+#  This class will test the functionality of following drivers
+#  1. Hardware monitoring sensors(hwmon driver) using sensors utility
+
+import time
+import subprocess
+import re
+
+from OpTestBMC import OpTestBMC
+from OpTestIPMI import OpTestIPMI
+from OpTestConstants import OpTestConstants as BMC_CONST
+from OpTestError import OpTestError
+from OpTestLpar import OpTestLpar
+from OpTestUtil import OpTestUtil
+
+
+class OpTestSensors():
+    #  Initialize this object
+    #  @param i_bmcIP The IP address of the BMC
+    #  @param i_bmcUser The userid to log into the BMC with
+    #  @param i_bmcPasswd The password of the userid to log into the BMC with
+    #  @param i_bmcUserIpmi The userid to issue the BMC IPMI commands with
+    #  @param i_bmcPasswdIpmi The password of BMC IPMI userid
+    #  @param i_ffdcDir Optional param to indicate where to write FFDC
+    #
+    # "Only required for inband tests" else Default = None
+    # @param i_lparIP The IP address of the LPAR
+    # @param i_lparuser The userid to log into the LPAR
+    # @param i_lparPasswd The password of the userid to log into the LPAR with
+    #
+    def __init__(self, i_bmcIP, i_bmcUser, i_bmcPasswd,
+                 i_bmcUserIpmi, i_bmcPasswdIpmi, i_ffdcDir=None, i_lparip=None,
+                 i_lparuser=None, i_lparPasswd=None):
+        self.cv_BMC = OpTestBMC(i_bmcIP, i_bmcUser, i_bmcPasswd, i_ffdcDir)
+        self.cv_IPMI = OpTestIPMI(i_bmcIP, i_bmcUserIpmi, i_bmcPasswdIpmi,
+                                  i_ffdcDir)
+        self.cv_LPAR = OpTestLpar(i_lparip, i_lparuser, i_lparPasswd)
+        self.util = OpTestUtil()
+
+    def test_hwmon_driver(self):
+        # This function will cover following
```

Comment:
Thanks for your suggestions. We will follow the same way as you mentioned in the above. We will use common directory for adding the more common functions which will be utilizing in test cases. And we will create a new directory ../op-test-framework/testcases for adding new test cases. Which will be useful for making regression test cases.  I will modify the scripts and re-send the patch again.